src/libsystemd/sd-device/device-private.c

```diff
@@ -756,9 +757,12 @@ int device_shallow_clone(sd_device *old_device, sd_device **new_device) {
         if (r < 0)
                 return r;
 
-        r = device_set_subsystem(ret, old_device->subsystem);
-        if (r < 0)
-                return r;
+        if (sd_device_get_subsystem(old_device, &subsystem) >= 0) {
+                r = device_set_subsystem(ret, old_device->subsystem);
```

Comment:
Shouldn't this use `subsystem`? If we have the encapsulation of get_subsystem(), I think we should use the return value.

Reply:
Ouch!

Reply:
Fixed.